src/api/app/controllers/source/errors.rb

```diff
@@ -94,4 +94,8 @@ class ProjectCopyNoPermission < APIError
   class NotLocked < APIError; end
 
   class InvalidFlag < APIError; end
+
+  class StagingProjectNotFound < APIError
+    setup 'not_found', 404
```

Comment:
btw: if you just setup 404, the error will be `staging_project_not_found` - and this is basically the only benefit of not using the generic `ActiveRecord` exception. The class name of the exception becomes the user facing error.

Reply:
Really, leaving the 'not_found' improves the API usage - a little bit. API errors follow e.g. this format:

```
osc -d api -X PUT -T Rakefile /source/GNOME:Factory/gnome-shell/Rakefile
...
<status code="put_file_no_permission">
   <summary>Insufficient permissions to store file in package gnome-shell, project GNOME:Factory</summary></status>
```

This maps to `raise PutFileNoPermission, "Insufficient permissions to store file in package #{@package_name}, project #{@project_name}"`

So with 'not_found', it says code='not_found', without it it says code='staging_workflow_not_found' or code='staging_project_not_found' - and the client has an easy way to differ. And you have to type less.

But I don't think it's important enough to hold it back more.